### DISCUSSION TITLE
能不能用python写插件啊，不会c#

### DISCUSSION THREAD

**Discussion by wjj-8283** [ClassIsland/ClassIsland#574] [Q&A] [answered]
c语言那一个系列代码又难敲又难理解，而且教室电脑上面跑vs卡的跟屎一样，但是python相对简单且不需要vs这种大型的ide，只需要一个轻量的vscode，所以能不能用python写啊？
如果不能，希望能有大佬开发出转译插件，将python代码转译成ci的pluginsdk的功能

**Comment by HelloWRC** [comment, marked as answer]
目前 ClassIsland 只能加载由 C# 或 VB.NET 等语言开发的插件，不能直接加载 Python 脚本。

目前远期有计划开发插件，让 ClassIsland 可以加载 Python 脚本，并通过 Python 调用 ClassIsland 的一些 API。这个插件的目标是实现一种更轻量的扩展方式，有点类似于浏览器中扩展和用户脚本的关系。

**Reply by MKStoler4096** [reply]
莫不成要叫 `ClassIDEA` ？

**Reply by wjj-8283** [reply]
感谢解答

**Reply by FrankXiDong** [reply]
1.7还是1.8实现（误）

**Comment by MKStoler4096** [comment]
实在不行单独准备一个 `com` 组件用来引用也行，这样的话就具有普适性了（）

**Reply by MKStoler4096** [reply]
引用一下这段话：
> ### 2.1 平台语言无关性
> 
>   COM实际上并不局限于Windows平台以及C++语言，因为COM它只是制定了一套通用的组件开发规范，这些规范并没有与具体的语言和平台进行绑定，Windows下的COM只是此标准的一种实现。
> 
>   COM是一套跨平台以及跨语言的组件开发规范，不管用户使用何种语言开发组件，也不管该组件运行在何种平台，只要开发的组件符合COM规范的要求，那么其它的应用程序就能够基于COM规范来正确的调用它。

**Comment by wjj-8283** [comment]
重新倒回来看这个discussion，好像C#是可以调用Python的语法的……只是有点麻烦

**Reply by doudou0720** [reply]
调用python还要有一个解释器，不如调用javascript

**Comment by doudou0720** [comment]
看了一下，或许可以用IronPython？

**Reply by doudou0720** [reply]
就是目前只支持到3.4，但是无缝衔接。
@wjj-8283

**Reply by doudou0720** [reply]
高版本可以用Python.NET

**Reply by Jack-tendy-538** [reply]
以我俗见，用python写跨.net插件需要解决启动问题与调用问题
启动的话，用IPC够用了
但是调用.net库就不好，首先要pythonnet，除非ci本体有动态库的api暴露，这样用ctypes都可以了
[比方说这样](https://github.com/Jack-tendy-538/ipa)

**Comment by doudou0720** [comment]
我~~撤回之前的言论~~觉得ZeroMQ还可以
